src/EditorFeatures/Core/Shared/Preview/PreviewWorkspace.cs

```diff
@@ -27,8 +27,7 @@ public PreviewWorkspace(HostServices hostServices)
         public PreviewWorkspace(Solution solution)
             : base(solution.Workspace.Services.HostServices, WorkspaceKind.Preview)
         {
-            var oldSolution = this.CurrentSolution;
-            var newSolution = this.SetCurrentSolution(solution);
+            var (oldSolution, newSolution) = this.SetCurrentSolution(solution);
 
             this.RaiseWorkspaceChangedEventAsync(WorkspaceChangeKind.SolutionChanged, oldSolution, newSolution);
```

Comment:
Curious to me we're raising the workspace changed event here when the type hasn't even left the constructor...not sure anybody could have subscribed yet?

Reply:
lol.  too true.

Reply:
ok.  so this isn't entirely true.  this just registers the event to be issued later.  and the registration happens through MEF, so it is possible for this to have impact.  it's definitely suspect though, and we should probably remove later.  but i'm not going to touch for now.

Reply:
The event is issued later, but the event handlers are captured synchronously; this was to avoid surprises where when subscribing you might be getting "old" events or unsubscribing might lose events you expected to see.

Reply:
> The event is issued later, but the event handlers are captured synchronously; 

the event handlers are retrieved through *mef*.  IT's not an actual C# event subscription for all of them.   :)